tritium-processor/src/main/java/com/palantir/tritium/processor/TritiumAnnotationProcessorStrategy.java

```diff
@@ -188,11 +189,17 @@ public List<FieldSpec> additionalFields(AdditionalFieldsArguments arguments) {
 
     @Override
     public void customize(CustomizeArguments arguments, TypeSpec.Builder generatedType) {
-        if (isDeprecated(arguments.type().type())) {
+        TypeElement type = arguments.type().type();
+        if (isDeprecated(type)) {
             generatedType.addAnnotation(Deprecated.class);
         }
 
-        TypeName annotatedTypeName = TypeName.get(arguments.type().type().asType());
+        SuppressWarnings suppressWarnings = type.getAnnotation(SuppressWarnings.class);
```

Comment:
This should probably be `isDeprecated`, right? Otherwise it forces users who want to instrument external interfaces with deprecated methods to define their own interface just so they can add `@SuppressWarnings`.

I think we'd also want to check for `@SuppressWarnings` on the method itself, possibly instead of the class level annotation.

Reply:
The case I'm hitting is an external interface that itself is not deprecated, but some returns types that are deprecated. If you're instrumenting an external type, you already have to declare your own interface, so I'm somewhat in favor of explicitly needing to also `@SuppressWarnings` so folks acknowledge the underlying type has deprecated types.

The code above on line 192-194 already propagates `@Deprecated` on an `@Instrument`ed type (see comment below with linked tests & output).

Reply:
> If you're instrumenting an external type, you already have to declare your own interface

Oh duh, I was being silly.

> so I'm somewhat in favor of explicitly needing to also @SuppressWarnings so folks acknowledge the underlying type has deprecated types

👍